File: AC/ARC/106/b.go
package main

import "fmt"

type Graph map[int][]int

func NewGraph() Graph {
	return Graph{}
}

func (g Graph) AddVertex(v int) {
	g[v] = []int{}
}

func (g Graph) AddEdge(v1, v2 int) {
	g[v1] = append(g[v1], v2)
	g[v2] = append(g[v2], v1)
}

func (g Graph) Graph() map[int][]int {
	return g
}

func (g Graph) Entry(v int) []int {
	return g[v]
}

func main() {
	var n, m, c, d int
	fmt.Scan(&n, &m)
	a := make([]int, n)
	b := make([]int, n)
	diff := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	for i := 0; i < n; i++ {
		fmt.Scan(&b[i])
		diff[i] = b[i] - a[i]
	}

	graph := NewGraph()
	for i := 0; i < n; i++ {
		graph.AddVertex(i)
	}
	for i := 0; i < m; i++ {
		fmt.Scan(&c, &d)
		graph.AddEdge(c-1, d-1)
	}
	fmt.Println(graph.Graph())
}

File: AC/ARC/106/b.py
class Graph:
    def __init__(self):
        self.dict = {}

    def add_vertex(self, v):
        self.dict[v] = []

    def add_edge(self, v1, v2):
        self.dict[v1].append(v2)
        self.dict[v2].append(v1)

    def graph(self):
        return self.dict

    def entry(self, v1):
        return self.dict[v1]


n, m = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

diff = [b[i]-a[i] for i in range(n)]


graph = Graph()
for i in range(n):
    graph.add_vertex(i)

for i in range(m):
    tmpc, tmpd = map(int, input().split())
    graph.add_edge(tmpc-1, tmpd-1)

for i in range(n):
    if diff[i] == (-1)*sum(diff[i] for i in graph.entry(i)):
        print("Yes")
        exit()
print("No")
